lib/Assertion/ParameterProvider.php

```diff
@@ -47,4 +49,15 @@ private function buildVariantData(Variant $variant)
 
         return $data;
     }
+
+    /**
+     * @return array<string,mixed>
+     */
+    private function subjectData(Subject $subject): array
+    {
+        return [
+            'time_unit' => $subject->getOutputTimeUnit(),
+            'time_precision' => $subject->getOutputTimePrecision(),
```

Comment:
ideally I think having tabular data would be better, this would enable the reports, assertions and progress loggers to operate on the same data structure